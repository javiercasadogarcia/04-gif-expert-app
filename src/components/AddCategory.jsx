import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({/*setCategories*/onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')    

    const onInputChange = (event) => {
        //console.log(event.target.value);
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        //console.log(event);
        event.preventDefault();
        //console.log(inputValue);
        if (inputValue.trim().length <= 1) return;

        //setCategories(categories => [inputValue, ...categories]);
        onNewCategory(inputValue);
        setInputValue('');
    }

  return (
    <>
        <form aria-label="form" onSubmit={onSubmit}>
            <input 
                type="text" 
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    </>
  )
}

AddCategory.prototypes = {
    onNewCategory: PropTypes.func.isRequired
}