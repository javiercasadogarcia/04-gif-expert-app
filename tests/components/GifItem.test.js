import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('GifItem tests', () => { 
    // arrange
    const title = 'Title'
    const url = 'http://url.com/'

    test('should match with the snapshot', () => { 
        // act
        const {container} = render(<GifItem title={title} url={url} />);

        // assert
        expect(container).toMatchSnapshot();
    });

    test('should show the image with the title and url', () => { 
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should show the title in the parragraph', () => { 
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    })
 })