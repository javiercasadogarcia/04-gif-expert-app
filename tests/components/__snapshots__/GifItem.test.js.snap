// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="http://url.com/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
